commons/src/main/java/info/metadude/android/eventfahrplan/commons/SystemClock.kt

```diff
@@ -0,0 +1,112 @@
+package info.metadude.android.eventfahrplan.commons
+
+import org.threeten.bp.*
+import org.threeten.bp.temporal.ChronoField
+
+
+/**
+ * An instance represents a moment in time. All operations are UTC based.
+ * Timezone based dates are retrieved using {#toZonedDateTime}.
+ *
+ * E.g.
+ *
+ * > SystemClock().toZonedDateTime(ZoneOffset.of("GMT+1"))
+ * */
+class SystemClock() : Clock {
+
+    private var time: Instant = Instant.now(org.threeten.bp.Clock.systemUTC())
+    private val utcZone = ZoneOffset.UTC
+
+	private constructor(instant: Instant) : this() {
+		time = instant
+	}
+
+	constructor(milliseconds: Long) : this() {
+        setToMilliseconds(milliseconds)
+    }
+
+	/**
+     * @param UTCDate must be in ISO-8601 format, e.g. "yyyy-MM-dd"
+	 * */
+    constructor(UTCDate: String) : this(LocalDate.parse(UTCDate).atStartOfDay().toInstant(ZoneOffset.UTC))
+
+    constructor(date: ZonedDateTime) : this(date.withZoneSameInstant(ZoneOffset.UTC).toInstant())
+
+    override val year: Int
+        get() = time.atZone(utcZone).year
+
+	override val month: Int
+        get() = time.atZone(utcZone).monthValue
+
+	override val monthDay: Int
+        get() = time.atZone(utcZone).dayOfMonth
+
+	override val hour: Int
+        get() = time.atZone(utcZone).hour
+
+	override val minute: Int
+        get() = time.atZone(utcZone).minute
+
+    override val minuteOfDay: Int
+        get() = time.atZone(utcZone).get(ChronoField.MINUTE_OF_DAY);
+
+    override fun setToNow() {
+        time = Instant.now(org.threeten.bp.Clock.systemUTC())
+	}
+
+    override fun startOfDay(): SystemClock {
+        return SystemClock(toUTCDateTime().toLocalDate().atStartOfDay().toInstant(utcZone))
+    }
+
+    override fun endOfDay(): SystemClock {
+        return SystemClock(toUTCDateTime().toLocalDate().atTime(LocalTime.MAX).toInstant(utcZone))
+    }
+
+    override fun setToMilliseconds(milliseconds: Long) {
+		time = Instant.ofEpochMilli(milliseconds)
+    }
+
+	override fun toMilliseconds() = time.toEpochMilli()
+
+    override fun toUTCDateTime(): LocalDateTime {
+        return time.atZone(utcZone).toLocalDateTime()
+	}
+
+    override fun toZonedDateTime(timeZoneOffset: ZoneOffset): ZonedDateTime {
+        return time.atZone(timeZoneOffset)
+    }
+
+    override fun minusHours(hours: Int) {
+        time = time.minusSeconds((hours * 3600).toLong())
```

Comment:
Instant.minusMinutes does not exist. Delegating to this.minusMinutes has no advantage to me, since it would just add another method call to the stack.

Reply:
Agreed. - I would have thought that the following would work but the corresponding tests fail:

``` kotlin
fun minusHours(hours: Int) {
    time.minus(hours.toLong(), ChronoUnit.HOURS)
}

fun minusMinutes(minutes: Int) {
    time.minus(minutes.toLong(), ChronoUnit.MINUTES)
}
```

Do you see why?

Reply:
Maybe ` time = time.minus(hours.toLong(), ChronoUnit.HOURS)` ?

Reply:
LOL. The classic mistake. Yes, with the assignment it works. Same for `plusMinutes`.